### PR TITLE
Hz 0216 update user password

### PR DESCRIPTION
**Pull request by Kelozheng**
1. save the updated to database.
2. update the alert function when the serve down (status code 500?)

### CODE REVIEW

File: src/pages/password-reset.tsx [hunk 1]
```diff
@@ -108,39 +122,61 @@ const validationSchema = yup.object({
...
-    onSubmit: (value) => {
-      console.log(value);
+    onSubmit: async ({ password }) => {
+      try {
+        if (router.query.code !== undefined) {
+          const token = router.query.code.toString();
+          const fileDataProcessed = Buffer.from(token, "base64").toString("binary");
+          const emailString = JSON.stringify(JSON.parse(fileDataProcessed).email);
+          const email = emailString.replace('"', "").replace('"', "");
+          axiosClient.patch("/agents/reset-password", { email, password });
```
Comment (yuanningd) [inline src/pages/password-reset.tsx]: Do you think we need await here?

File: src/pages/password-reset.tsx [hunk 2]
```diff
@@ -108,39 +122,61 @@ const validationSchema = yup.object({
...
+          router.push("/password-reset-success");
+        }
+      } catch (error) {
+        if (error instanceof AxiosError && error.response?.status === 500) {
+          setAccountActive({ severity: "error", display: "flex", text: "Wait for server respond" });
```
Comment (yuanningd) [inline src/pages/password-reset.tsx]: If server responds with 500, why would you ask user to wait?

File: src/pages/password-reset.tsx [hunk 3]
```diff
@@ -108,39 +122,61 @@ const validationSchema = yup.object({
...
   const formik = useFormik({
     initialValues: {
       password: ""
     },
     validationSchema,
-    onSubmit: (value) => {
-      console.log(value);
+    onSubmit: async ({ password }) => {
+      try {
+        if (router.query.code !== undefined) {
```
Comment (yuanningd) [inline src/pages/password-reset.tsx]: If(router.query.code)

**Comment by yuanningd** [comment]
Where’s the test?!

**Review by Timguau — APPROVED** [review]

**Review by jinzhih — APPROVED** [review]

**Pull request merged**